packages/assets-controllers/src/MultichainAssetsController/MultichainAssetsController.test.ts

```diff
@@ -0,0 +1,1539 @@
+import { ControllerMessenger } from '@metamask/base-controller';
+import type {
+  AccountAssetListUpdatedEventPayload,
+  CaipAssetTypeOrId,
+} from '@metamask/keyring-api';
+import { EthAccountType, EthMethod, EthScope } from '@metamask/keyring-api';
+import { KeyringTypes } from '@metamask/keyring-controller';
+import type { InternalAccount } from '@metamask/keyring-internal-api';
+import type { PermissionConstraint } from '@metamask/permission-controller';
+import type { SubjectPermissions } from '@metamask/permission-controller';
+import type { Snap } from '@metamask/snaps-utils';
+import { useFakeTimers } from 'sinon';
+import { v4 as uuidv4 } from 'uuid';
+
+import type {
+  AssetMetadataResponse,
+  MultichainAssetsControllerMessenger,
+  MultichainAssetsControllerState,
+} from './MultichainAssetsController';
+import {
+  getDefaultMultichainAssetsControllerState,
+  MultichainAssetsController,
+} from './MultichainAssetsController';
+import { advanceTime } from '../../../../tests/helpers';
+import type {
+  ExtractAvailableAction,
+  ExtractAvailableEvent,
+} from '../../../base-controller/tests/helpers';
+
+const mockSolanaAccount = {
+  type: 'solana:data-account',
+  id: 'a3fc6831-d229-4cd1-87c1-13b1756213d4',
+  address: 'EBBYfhQzVzurZiweJ2keeBWpgGLs1cbWYcz28gjGgi5x',
+  options: {
+    scope: 'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp',
```

Comment:
Also, updating from mainnet to devnet to match your other mocked values:

```suggestion
  scopes: [SolScope.Devnet]
  options: {
    scope: SolScope.Devnet,
```